api/src/mutations/mint.rs

```diff
@@ -126,18 +105,35 @@ impl Mutation {
             },
         };
 
+        // insert a collection mint record into database
         let collection_mint_active_model = collection_mints::ActiveModel {
             collection_id: Set(collection.id),
             address: Set(mint_address.to_string()),
-            owner: Set(input.recipient),
+            owner: Set(input.recipient.clone()),
             creation_status: Set(CreationStatus::Pending),
             created_by: Set(user_id),
             ..Default::default()
         };
 
         let collection_mint_model = collection_mint_active_model.insert(conn).await?;
+
+        // inserts a purchase record in the database
+        let purchase_am = purchases::ActiveModel {
+            mint_id: Set(collection_mint_model.id),
+            customer_id: Set(user_id),
```

Comment:
The user_id is either the contact of the auth token or the org member. Its not the customer.

I think we can put something together were we find a customer by its wallet address so lets drop customer_id from the purchases table.